controller/lib/hal/hal_stm32_regs.h

```diff
@@ -0,0 +1,444 @@
+    REG gif2 : 1;  // global interrupt flag
+    REG tcif2 : 1; // transfer complete (TC) flag
+    REG htif2 : 1; // half transfer (HT) flag
+    REG teif2 : 1; // transfer error (TE) flag
+    REG gif3 : 1;  // global interrupt flag
+    REG tcif3 : 1; // transfer complete (TC) flag
+    REG htif3 : 1; // half transfer (HT) flag
+    REG teif3 : 1; // transfer error (TE) flag
+    REG gif4 : 1;  // global interrupt flag
+    REG tcif4 : 1; // transfer complete (TC) flag
+    REG htif4 : 1; // half transfer (HT) flag
+    REG teif4 : 1; // transfer error (TE) flag
+    REG gif5 : 1;  // global interrupt flag
+    REG tcif5 : 1; // transfer complete (TC) flag
+    REG htif5 : 1; // half transfer (HT) flag
+    REG teif5 : 1; // transfer error (TE) flag
+    REG gif6 : 1;  // global interrupt flag
+    REG tcif6 : 1; // transfer complete (TC) flag
+    REG htif6 : 1; // half transfer (HT) flag
+    REG teif6 : 1; // transfer error (TE) flag
+    REG gif7 : 1;  // global interrupt flag
+    REG tcif7 : 1; // transfer complete (TC) flag
+    REG htif7 : 1; // half transfer (HT) flag
+    REG teif7 : 1; // transfer error (TE) flag
+    REG rsvd : 4;
+  } intStat; // interrupt status register
+
+  struct {
+    REG gif1 : 1;  // global interrupt flag
+    REG tcif1 : 1; // transfer complete (TC) flag
+    REG htif1 : 1; // half transfer (HT) flag
+    REG teif1 : 1; // transfer error (TE) flag
+    REG gif2 : 1;  // global interrupt flag
+    REG tcif2 : 1; // transfer complete (TC) flag
+    REG htif2 : 1; // half transfer (HT) flag
+    REG teif2 : 1; // transfer error (TE) flag
+    REG gif3 : 1;  // global interrupt flag
+    REG tcif3 : 1; // transfer complete (TC) flag
+    REG htif3 : 1; // half transfer (HT) flag
+    REG teif3 : 1; // transfer error (TE) flag
+    REG gif4 : 1;  // global interrupt flag
+    REG tcif4 : 1; // transfer complete (TC) flag
+    REG htif4 : 1; // half transfer (HT) flag
+    REG teif4 : 1; // transfer error (TE) flag
+    REG gif5 : 1;  // global interrupt flag
+    REG tcif5 : 1; // transfer complete (TC) flag
+    REG htif5 : 1; // half transfer (HT) flag
+    REG teif5 : 1; // transfer error (TE) flag
+    REG gif6 : 1;  // global interrupt flag
+    REG tcif6 : 1; // transfer complete (TC) flag
+    REG htif6 : 1; // half transfer (HT) flag
+    REG teif6 : 1; // transfer error (TE) flag
+    REG gif7 : 1;  // global interrupt flag
+    REG tcif7 : 1; // transfer complete (TC) flag
+    REG htif7 : 1; // half transfer (HT) flag
+    REG teif7 : 1; // transfer error (TE) flag
+    REG rsvd : 4;
+  } intClr; // interrupt flag clear register
+  struct {
+    struct {
+      REG enable : 1;   // channel enable
+      REG tcie : 1;     // transfer complete interrupt enable
+      REG htie : 1;     // half transfer interrupt enable
+      REG teie : 1;     // transfer error interrupt enable
+      REG dir : 1;      // data xfer direction 0: per->mem, 1: mem->per
+      REG circular : 1; // circular mode
```

Comment:
Perhaps add an enum for direction.  That would make it even more clear what a dir of 0 vs 1 means.

Reply:
Done, also added bit size enum, everything else looks pretty self explanatory.